zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java

```diff
@@ -1457,7 +1457,7 @@ public void afterStatusChange(Job job, Status before, Status after) {
         LOG.info("Job {} is finished", job.getId());
         try {
           //TODO(khalid): may change interface for JobListener and pass subject from interpreter
-          note.persist(null);
+          note.persist(job instanceof Paragraph ? ((Paragraph) job).getAuthenticationInfo() : null);
```

Comment:
@jongyoul can you give little more details on when `Job` would be an instance of `Paragraph` and otherwise